homeassistant/components/twinkly/light.py

```diff
@@ -321,14 +331,55 @@ async def async_update(self) -> None:
 
     async def async_update_movies(self) -> None:
         """Update the list of movies (effects)."""
-        movies = await self._client.get_saved_movies()
-        _LOGGER.debug("Movies: %s", movies)
-        if movies and "movies" in movies:
-            self._movies = movies["movies"]
+        try:
+            movies = self._valid_twinkly_response(
+                await self._client.get_saved_movies(), check_for="movies"
+            )
+        except TwinklyError as error:
+            _LOGGER.warning(error)
+            return
+        self._movies = movies["movies"]
 
     async def async_update_current_movie(self) -> None:
         """Update the current active movie."""
-        current_movie = await self._client.get_current_movie()
-        _LOGGER.debug("Current movie: %s", current_movie)
-        if current_movie and "id" in current_movie:
-            self._current_movie = current_movie
+        try:
+            self._current_movie = self._valid_twinkly_response(
+                await self._client.get_current_movie()
+            )
+        except TwinklyError as error:
+            _LOGGER.warning(error)
+
+    async def async_update_current_color(self) -> None:
+        """Update the current active color."""
+        try:
+            current_color = self._valid_twinkly_response(
+                await self._client.get_current_colour()
+            )
+        except TwinklyError as error:
+            _LOGGER.warning(error)
+            return
+        if self._attr_color_mode == ColorMode.RGBW:
+            self._attr_rgbw_color = color_rgb_to_rgbw(
+                current_color["red"],
+                current_color["green"],
+                current_color["blue"],
+            )
+        elif self._attr_color_mode == ColorMode.RGB:
+            self._attr_rgb_color = (
+                current_color["red"],
+                current_color["green"],
+                current_color["blue"],
+            )
+
+    def _valid_twinkly_response(
+        self, response: dict[Any, Any], check_for: str | None = None
+    ) -> dict[Any, Any]:
+        """Validate twinkly-responses from the API."""
+        if (
+            response
+            and response.get(TWINKLY_RETURN_CODE) == TWINKLY_RETURN_CODE_OK
+            and (not check_for or check_for in response)
+        ):
+            _LOGGER.debug("Twinkly response: %s", response)
+            return response
+        raise TwinklyError(f"Invalid response from Twinkly: {response}")
```

Comment:
I think it makes more sense if we check if we can raise it upstream in the library instead of wrapping it here.

That said, if we need to wrap this, I suggest creating a local class that extends the client to wrap its methods (instead of this proxy method).

../Frenck

Reply:
I agree that it would be best if upstream was a bit less "transparent", and did some more validation.
For now, I have added a simple wrapper class.